src/packageurl/contrib/url2purl.py

```diff
@@ -276,3 +276,23 @@ def build_sourceforge_purl(uri):
 @purl_router.route(cargo_pattern)
 def build_cargo_purl(uri):
     return purl_from_pattern('cargo', cargo_pattern, uri)
+
+
+@purl_router.route('https?://github.com.*/.*')
```

Comment:
What about the `www.` that sometimes appears for these github urls?

Reply:
I refered it from purl-spec, it's given that https://github.com is the default repository for GitHub, so what say ?

Reply:
Yeah you might be right, it looks like www.github.com gets redirected to github.com

Reply:
I agree with @majurg .. we want to accept inputs in a flexible way. In fact this is the whole purpose there: take possibly messy URLs and turn them in a clean purl.
also there could be many other cases beyond www prefix?
What about `https://raw.githubusercontent.com/volatilityfoundation/dwarf2json/master/LICENSE.txt` or API URLs or  `https://codeload.github.com/nexB/scancode.io/tar.gz/1.0` 

So for clarity we should have multiple patterns here.

 Also even the current form may not be correct. Why `.*` right after `.com`? and the dot in `.com`?
This looks better imho (but we still ned to accommodate more  as stated above)
`@purl_router.route('https?://github\.com/.*')`

Reply:
Hi, can you provide me some more example URLs for codeload and API ones, I have understood the raw.githubusercontent, but need some sample URLs for codeload and API ones :)

Reply:
- https://api.github.com/repos/nexB/scancode-toolkit/commits/40593af0df6c8378d2b180324b97cb439fa11d66
- https://api.github.com/repos/nexB/scancode-toolkit/

- for codeload when you fetch https://github.com/nexB/scancode-toolkit/archive/v3.1.1.zip (which is another url) or https://github.com/nexB/scancode-toolkit/archive/v3.1.1.tar.gz  the url is redirected to https://codeload.github.com/nexB/scancode-toolkit/zip/v3.1.1 or https://codeload.github.com/nexB/scancode-toolkit/zip/v3.1.1 or https://codeload.github.com/nexB/scancode-toolkit/tar.gz/v3.1.1

These codeload URL are 404 after a while, but they could still exists so we should try to craft a pulr

Reply:
Thanks :D